controllers/confidential_handler.go

```diff
@@ -92,12 +97,19 @@ func (r *KataConfigOpenShiftReconciler) handleConfidentialPeerPods(state Feature
 
 	// Patch peer pods configMap, if it exists.
 	var peerpodsCMData map[string]string
+	var keysToRemove map[string]string
 	if state == Enabled {
 		peerpodsCMData = map[string]string{"DISABLECVM": "false"}
+		// Remove INITDATA if it matches the default value
+		keysToRemove = map[string]string{"INITDATA": defaultNonCCInitdata}
```

Comment:
This will silently go against the intent of the user if they want to set such a permissive INITDATA on purpose... is it something we want ?

As said earlier, this is an ownership problem. It doesn't look right to tell the user they can set INITDATA if OSC is likely to overwrite it. It looks like user input should go to a dedicated CM or at least to a separate CUSTOM_INITDATA entry in the peerpods CM.

Cc @pmores for insights.